### PR TITLE
fix https://github.com/qznc/d-money/issues/11

### PR DESCRIPTION
**Pull request by mw66**
fixed the ctor, add tests, all passing.

### CODE REVIEW

**Comment by mw66** [comment]
https://forum.dlang.org/post/itasjnvfqwwiorchttcb@forum.dlang.org

there's also parsing that is not super compliant

EUR y = EUR("1.000.000,00");
writeln(y);                     // F!

=====================================
The original parsing regex used internally is a very simple one:
https://github.com/qznc/d-money/blob/master/source/money.d#L98

I don't want to enhanced it to be a full sscanf().

This is just a maintenance bug fix.

File: source/money.d [hunk 1]
```diff
@@ -315,6 +349,17 @@ struct currency(string currency_name, int dec_places = 4, roundingMode rmode = r
...
     }
 
+    int opCmp(T)(const T other) const if (isNumeric!T)
+    {
+        if (this.amount < other * __factor)
+            return -1;
+        else if (this.amount > other * __factor)
+            return 1;
+        else
+            return 0;
```
Comment (qznc) [inline source/money.d:365]: The return 0 case is not covered by the unittest.

**Review by qznc — CHANGES_REQUESTED** [review]
Sorry, I'm not checking Github that often anymore. You changes look fine and I would like to include merge them.

Just one more request: This library is pride of 100% coverage and currently one line missing. :)

File: source/money.d (hunk 1 above)
Comment (mw66) [reply source/money.d:365]: done.

**Pull request merged**